packages/edit-site/src/components/left-sidebar/navigation-panel/index.js

```diff
@@ -56,25 +58,37 @@ const NavigationPanel = () => {
 						? `template-${ templateId }`
 						: `template-part-${ templatePartId }`
 				}
+				onActivateMenu={ setActiveMenu }
 			>
-				<NavigationBackButton
-					backButtonLabel={ __( 'Dashboard' ) }
-					className="edit-site-navigation-panel__back-to-dashboard"
-					href="index.php"
-					ref={ ref }
-				/>
+				{ activeMenu === 'root' && (
+					<NavigationBackButton
+						backButtonLabel={ __( 'Dashboard' ) }
+						className="edit-site-navigation-panel__back-to-dashboard"
+						href="index.php"
+						ref={ ref }
+					/>
+				) }
+
+				<NavigationMenu title="Theme">
+					<NavigationItem
+						title="Templates"
+						navigateToMenu="templates"
+					/>
 
-				<NavigationMenu title="Home">
-					<TemplateSwitcher
-						page={ page }
-						activeId={ templateId }
-						onActiveIdChange={ setTemplate }
+					<NavigationItem
+						title="Template parts"
+						navigateToMenu="template-parts"
+					/>
+
+					<TemplatesMenu onActiveIdChange={ setTemplate } />
+
+					<TemplatePartsMenu
 						onActiveTemplatePartIdChange={ setTemplatePart }
-						onAddTemplate={ addTemplate }
-						onRemoveTemplate={ removeTemplate }
 					/>
 				</NavigationMenu>
 			</Navigation>
+
```

Comment:
Now that we have templates and template parts in separate menus, how about simplifying the API and have `onActivate` (or `onActivateItem`) in both `TemplatesMenu` and `TemplatePartsMenu`, instead of `onActiveIdChange` and `onActiveTemplatePartIdChange`?